### PR TITLE
DOC: Minor improvements groupby user guide

### PR DESCRIPTION
**Pull request by rhshadrach**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/user_guide/groupby.rst [hunk 1]
```diff
@@ -846,15 +845,14 @@ The following methods on GroupBy act as transformations. Of these methods, only
         :meth:`~.DataFrameGroupBy.cumsum`;Compute the cumulative sum within each group
         :meth:`~.DataFrameGroupBy.diff`;Compute the difference between adjacent values within each group
         :meth:`~.DataFrameGroupBy.ffill`;Forward fill NA values within each group
-        :meth:`~.DataFrameGroupBy.fillna`;Fill NA values within each group
```
Comment (rhshadrach) [inline doc/source/user_guide/groupby.rst:849]: Removed since fillna is deprecated

File: doc/source/user_guide/groupby.rst [hunk 2]
```diff
@@ -1308,17 +1306,8 @@ is only needed over one column (here ``colname``), it may be filtered
...
-
-    # Decimal columns can be sum'd explicitly by themselves...
     df_dec.groupby(["id"])[["dec_column"]].sum()
 
-    # ...but cannot be combined with standard data types or they will be excluded
-    df_dec.groupby(["id"])[["int_column", "dec_column"]].sum()
-
-    # Use .agg function to aggregate over standard and "nuisance" data types
-    # at the same time
-    df_dec.groupby(["id"]).agg({"int_column": "sum", "dec_column": "sum"})
```
Comment (rhshadrach) [inline doc/source/user_guide/groupby.rst:1320]: You used to need to do this to include nuisance columns; not anymore. Now they are included by default.

File: doc/source/user_guide/groupby.rst [hunk 3]
```diff
@@ -1350,35 +1339,53 @@ The returned dtype of the grouped will *always* include *all* of the categories
 
    s = (
        pd.Series([1, 1, 1])
-       .groupby(pd.Categorical(["a", "a", "a"], categories=["a", "b"]), observed=False)
+       .groupby(pd.Categorical(["a", "a", "a"], categories=["a", "b"]), observed=True)
```
Comment (rhshadrach) [inline doc/source/user_guide/groupby.rst:1342]: This demonstrates unobserved categories are being kept, so I think it makes more sense to show with `observed=True`

**Comment by mroeschke** [comment]
Just noting to feel free to liberally move examples and behavior notes to docstrings or remove sections that are too niche/outdated. IMO I feel that information is best served in docstrings as opposed to user guides given how outdated they are

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @rhshadrach

**Pull request merged**